dns-metrics.schema.json

```diff
@@ -130,6 +130,10 @@
                     "description": "The number of DNS queries and/or responses that was unexpected",
                     "type": "integer"
                 },
+                "discarded": {
+                    "description": "The number of DNS queries and/or responses that was discarded",
```

Comment:
Oh, this is an excellent point. I think we need counter for queries "not sent because it's garbage" and "not received because it's garbage". I would go for `discarded_dns_request` `discarded_dns_response` or something like that.

Note the `dns` - as mentioned somewhere in @libor-peltan-cznic's notes there are two types of "discarded" - DNS and non-DNS:
- DNS type is that something does not parse as DNS payload even though it should be a DNS payload - e.g. during PCAP replay a packet is too short to be a DNS message
- non-DNS type is that e.g. network stack (or driver if we are in AF_XFP land) refused to send stuff - e.g. because buffers were full

In general both counters should be zero all the time, but if they are not it's very important to distinguish the two cases - one indicates problem in test data and the other with test environment itself.

Reply:
I got confused by "counting counters", so here's what I meant:
- discarded_dns_request
- discarded_dns_response
- discarded_other_request
- discarded_other_response

Of course it can be somehow put into sub-objects etc. I think you get the idea.

Reply:
I think that garbage in input shouldn't be represented in these stats, since if your input has garbage you would always add those stats into this then. So, you should clean your input! Or we need an object about the input.

But splitting queries/responses for `discarded`, `timeouts`, `interrupted`, `unexpected` sure, but can I merge this and do that in another PR?